common/djangoapps/external_auth/views.py

```diff
@@ -555,7 +555,7 @@ def _safe_postlogin_redirect(redirect_to, safehost, default_redirect='/'):
     @param safehost: which host is safe to redirect to
     @return: an HttpResponseRedirect
     """
-    if is_safe_url(url=redirect_to, host=safehost):
+    if 'studio.suclass.stanford.edu' in redirect_to or is_safe_url(url=redirect_to, host=safehost):
```

Comment:
this is actually vulnerable to an attack such as redirect to `https://attacker.com/?urlparameter=studio.suclass.stanford.edu`, so you'll have to use urlparse to get the host parameter explicitly.

Also, we might as well do it right to begin with.  I think we should have a setting SHIB_REDIRECT_DOMAIN_WHITELIST
that is a dict, with the keys being the "from" domain and the "values" being the "to" domain.

So it'd look like the following for suclass.stanford.edu
```
SHIB_REDIRECT_DOMAIN_WHITELIST = {
    'suclass.stanford.edu': ['studio.suclass.stanford.edu']
}
```
and of course redirecting to yourself is always fine.